rust_src/src/buffers.rs

```diff
@@ -103,3 +103,12 @@ pub fn get_buffer(buffer_or_name: LispObject) -> LispObject {
         ))
     }
 }
+
+/// Return the current buffer as a Lisp object.
+#[lisp_fn]
+pub fn current_buffer() -> LispObject {
```

Comment:
LGTM. The only way I see to make it prettier would be to have `unsafe fn LispObject::from_ptr()`, but it wouldn't save much.